tests/components/bluesound/test_init.py

```diff
@@ -0,0 +1,47 @@
+"""Test __init__."""
+
+import asyncio
+
+from pyblu.errors import PlayerUnreachableError
+
+from homeassistant.components.bluesound import async_unload_entry
+from homeassistant.core import HomeAssistant
+
+from .conftest import PlayerMocks
+
+from tests.common import MockConfigEntry
+
+
+async def test_setup_entry(hass: HomeAssistant, setup_config_entry: None) -> None:
+    """Test a successful setup entry."""
+    assert hass.states.get("media_player.player_name1111").state == "playing"
+
+
+async def test_unload_entry(
+    hass: HomeAssistant, setup_config_entry: None, config_entry: MockConfigEntry
+) -> None:
+    """Test entries are unloaded correctly."""
+    assert await async_unload_entry(hass, config_entry)
+    await hass.async_block_till_done()
+
+    assert hass.states.get("media_player.player_name1111").state == "unavailable"
+
+
+async def test_unload_entry_while_offline(
+    hass: HomeAssistant,
+    setup_config_entry: None,
+    config_entry: MockConfigEntry,
+    player_mocks: PlayerMocks,
+) -> None:
+    """Test entries are unloaded correctly when the player is offline."""
```

Comment:
we should not unload entries when its offline

Reply:
What should be done when unload is called while the player is offline?

Reply:
Oh maybe I misinterpreted the name, it sounded like that if the player would become unavailable, it would unload the config entry, and that's not what we want

Reply:
Ah ok. It is there to ensure that the entry can be unloaded while the player is offline. I will change the test name to clarify.